silabs-multiprotocol/rootfs/etc/s6-overlay/s6-rc.d/otbr-agent/run

```diff
@@ -5,73 +5,79 @@
 
 . /etc/s6-overlay/scripts/otbr-agent-common
 
-declare backbone_if
 declare device
 declare baudrate
 declare flow_control
 declare otbr_log_level
 declare otbr_log_level_int
+declare otbr_infra_if
 declare otbr_rest_listen
 declare otbr_rest_listen_port
 
-backbone_if="$(bashio::api.supervisor 'GET' '/network/info' '' 'first(.interfaces[] | select (.primary == true)) .interface')"
-
 otbr_log_level=$(bashio::string.lower "$(bashio::config otbr_log_level)")
 case "${otbr_log_level}" in
     debug)
-        otbr_log_level_int="7"
+        otbr_log_level_int='7'
         ;;
     info)
-        otbr_log_level_int="6"
+        otbr_log_level_int='6'
         ;;
     notice)
-        otbr_log_level_int="5"
+        otbr_log_level_int='5'
         ;;
     warning)
-        otbr_log_level_int="4"
+        otbr_log_level_int='4'
         ;;
     error)
-        otbr_log_level_int="3"
+        otbr_log_level_int='3'
         ;;
     critical)
-        otbr_log_level_int="2"
+        otbr_log_level_int='2'
         ;;
     alert)
-        otbr_log_level_int="1"
+        otbr_log_level_int='1'
         ;;
     emergency)
-        otbr_log_level_int="0"
+        otbr_log_level_int='0'
         ;;
     *)
         bashio::exit.nok "Unknown otbr_log_level: ${otbr_log_level}"
         ;;
 esac
 
-if [ -z ${backbone_if} ]; then
-    bashio::log.warning "No primary network interface found! Using static eth0."
-    backbone_if="eth0"
+if bashio::config.has_value 'otbr_infra_if'; then
+    otbr_infra_if="$(bashio::config 'otbr_infra_if')"
+    bashio::log.info "Using configured network interface ${otbr_infra_if} for otbr_infra_if."
+else
+    otbr_infra_if="$(bashio::api.supervisor 'GET' '/network/info' '' 'first(.interfaces[] | select (.primary == true)) .interface')"
+    if [ -n "${otbr_infra_if}" ]; then
+        bashio::log.info "Using primary network interface ${otbr_infra_if} for otbr_infra_if."
+    else
+        bashio::log.warning 'No primary network interface found!  Using static eth0 for otbr_infra_if.'
+        otbr_infra_if='eth0'
```

Comment:
This is actually a bit legacy. `eth0` is almost never the right interface. It would be better if we exit here with an error, e.g.
```suggestion
        bashio::exit.nok  'No primary network interface found!'
```

Reply:
Agreed.  I thought this seemed like an odd choice for a default, but I didn't want to break anyone if this is actually used somewhere.  I will change it.